kuka_kr6_support/urdf/kr6r900_2_macro.xacro

```diff
@@ -0,0 +1,182 @@
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_1.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_2">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/visual/link_2.stl"/>
+        </geometry>
+        <xacro:material_kuka_ral_traffic_white />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_2.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_3">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/visual/link_3.stl" />
+        </geometry>
+        <xacro:material_kuka_ral_traffic_white />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_3.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_4">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/visual/link_4.stl" />
+        </geometry>
+        <xacro:material_kuka_ral_traffic_white />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_4.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_5">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/visual/link_5.stl" />
+        </geometry>
+        <xacro:material_kuka_ral_traffic_white />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_5.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_6">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/visual/link_6.stl" />
+        </geometry>
+        <xacro:material_kuka_black />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://kuka_kr10_support/meshes/kr10r900_2/collision/link_6.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <!-- This frame corresponds to the $FLANGE coordinate system in KUKA KRC controllers. -->
+    <link name="${prefix}tool0"/>
```

Comment:
Aren't there now two `tool0` links?

I'd assumed CI would've picked up on this, but it doesn't look like it.

Reply:
I believe I can explain this: the `load` test does run `xacro`, but `xacro` does not interpret the resulting urdf.

It may be nice to add a `check_urdf` or something similar to the files in `test/`.

Reply:
yes, exactly. As always when in doubt I looked at the fanuc packages but it seems they start rviz etc even in test mode?

Reply:
No. That's a `roslaunch_check` test. That just makes sure the `.launch` file can be parsed. So it will catch `xacro` parse errors, but not errors in the resulting URDF.

Reply:
wow, you even gave a lecture on checking urdf. I'll try to open a PR for that

Reply:
a lecture? :)

Reply:
while googling I found this: https://ocw.tudelft.nl/course-lectures/2-2-3-checking-for-correctness/

I haven't found anything yet that verifies URDF directly from the ros parameter server unfortunately. Maybe I can include it in `check_opw` (just empty promises so far, no code yet ...)

mind to approve/recheck?

Reply:
Ouch. Not my best work. Reading an auto-cue is not my forte.

Reply:
This might be helpful for checking / parsing the `.xacro`s at build / test: https://github.com/ros/xacro/issues/262.